### PR TITLE
Some image recipe updates

### PR DESCRIPTION
**Pull request by Villemoes**
While working on some qt recipes I was led to update some of our image recipes, as well as add one for libwebp. All three pass packageqa for me.

### CODE REVIEW

File: recipes/libgif/libgif.inc [hunk 1]
```diff
@@ -0,0 +1,26 @@
+DESCRIPTION = "GIFLIB is a package of portable tools and library routines for working with GIF images."
+HOMEPAGE = "http://giflib.sourceforge.net/"
+LICENSE = "MIT"
+
+SRC_URI = "http://kent.dl.sourceforge.net/project/giflib/giflib-${PV}.tar.bz2"
+
+inherit autotools library auto-package-utils
+
+S = "${SRCDIR}/giflib-${PV}"
```
Comment (kimrhh) [inline recipes/libgif/libgif.inc]: isn't that what S is by default?

File: recipes/libtiff/tiff.inc [hunk 2]
```diff
@@ -1,24 +1,61 @@
...
 LICENSE = "BSD-2-Clause*"
-HOMEPAGE = "http://www.remotesensing.org/libtiff/"
 
 RECIPE_TYPES = "machine native"
+SRC_URI = "ftp://download.osgeo.org/libtiff/tiff-${PV}.tar.gz"
 
 COMPATIBLE_HOST_ARCHS = ".*linux"
 
-SRC_URI = "ftp://ftp.remotesensing.org/pub/libtiff/tiff-${PV}.tar.gz"
+inherit autotools-autoreconf c++ auto-package-utils auto-package-libs
```
Comment (kimrhh) [inline recipes/libtiff/tiff.inc]: don't see any need for autoreconf here

File: recipes/libtiff/tiff.inc [hunk 3]
```diff
@@ -1,24 +1,61 @@
...
 
-inherit autotools c++
+RECIPE_FLAGS += "tiff_jpeg tiff_lzma tiff_zlib"
```
Comment (kimrhh) [inline recipes/libtiff/tiff.inc]: i don't see any configure handling of these, are they automagic ? would prefer a --disable/--without when they are disabled if that works with the provided configure

**Comment by kimrhh** [comment]
please rebase this onto current master, i can no longer easily find the triggers in the github webhooks thingy

File: recipes/libgif/libgif.inc (hunk 1 above)
Comment (kimrhh) [inline recipes/libgif/libgif.inc]: oh, do not name the recipe libgiff when upstream calls the software giflib, that just makes it harder to find stuff

**Comment by Villemoes** [comment]
All comments should be addressed. Now passing explicit --disable flags in libtiff, appending --enable if USE flag is set. autoreconf dropped, not sure why I added that. libgif -> giflib.

File: recipes/libwebp/webp.inc [hunk 4]
```diff
@@ -0,0 +1,17 @@
+
+SRC_URI = "https://storage.googleapis.com/downloads.webmproject.org/releases/webp/libwebp-${PV}.tar.gz"
+
+inherit auto-package-libs auto-package-utils autotools
+
```
Comment (kimrhh) [inline recipes/libwebp/webp.inc]: you'll have to disable mingw build in this, using the COMPATIBLE_HOST_ARCH dance

**Comment by sknsean** [comment]
@Villemoes try to add something like this in libtiff
#Only used for Aarch64
AUTOCONFDIRS = "/build/autoconf"

**Comment by Villemoes** [comment]
libtiff now builds on aarch64, at least for me. @kimrhh was this the COMPATIBLE_HOST_ARCHS thing I needed?

**Comment by kimrhh** [comment]
no that was because it also fails for mingw :)

**Comment by kimrhh** [comment]
no that was because it also fails for mingw :)

**Comment by kimrhh** [comment]
no that was because webp fails for mingw :)

**Pull request merged**